lib/jellyfish_web/socket.ex

```diff
@@ -14,65 +16,58 @@ defmodule JellyfishWeb.Socket do
 
   @impl true
   def connect(state) do
-    Logger.info("New incoming WebSocket connection...")
-
-    with {:ok, peer_id} <- Map.fetch(state.params, "peer_id"),
-         {:ok, room_id} <- Map.fetch(state.params, "room_id"),
-         {:ok, room_pid} <- RoomService.find_room(room_id),
-         {:ok, :disconnected} <- Room.get_peer_connection_status(room_pid, peer_id) do
-      state =
-        state
-        |> Map.put(:room_id, room_id)
-        |> Map.put(:peer_id, peer_id)
-        |> Map.put(:room_pid, room_pid)
-
-      Logger.info("""
-      WebSocket connection from peer #{inspect(peer_id)} accepted, \
-      room #{inspect(room_id)}
-      """)
-
-      {:ok, state}
-    else
-      {:ok, :connected} ->
-        Logger.warn("""
-        WebSocket connection for peer #{inspect(state.params["peer_id"])} in room \
-        #{inspect(state.params["room_id"])} already exists, rejected
-        """)
-
-        {:error, :already_connected}
+    Logger.info("New incoming WebSocket connection, accepting")
 
-      {:error, :room_not_found} ->
-        Logger.warn("""
-        Room #{inspect(state.params["room_id"])} not found, \
-        ignoring incoming WebSocket connection
-        """)
+    {:ok, state}
+  end
 
-        {:error, :room_not_found}
+  @impl true
+  def init(state) do
+    {:ok, Map.put(state, :authenticated?, false)}
+  end
 
-      {:error, :peer_not_found} ->
+  @impl true
+  def handle_in({encoded_message, [opcode: :text]}, %{authenticated?: false} = state) do
+    case Jason.decode(encoded_message) do
+      {:ok, %{"type" => "controlMessage", "data" => %{"type" => "authRequest", "token" => token}}} ->
+        with {:ok, %{peer_id: peer_id, room_id: room_id}} <- PeerToken.verify(token),
+             {:ok, room_pid} <- RoomService.find_room(room_id) do
+          Room.set_peer_connected(room_pid, peer_id)
```

Comment:
Shouldn't we handle return value from this call to set_peer_connected? We probably don't want to have two WebSocket connections opened to the same peer.